File: 2155-find-missing-observations/2155-find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int sum = 0;
        int m = rolls.size();
        for(int roll : rolls) {
            sum += roll;
        }

        int goal = (m + n) * mean;
        int left = goal - sum;

        if(left > 6 * n || left < n) return {};

        vector<int> ans (n, left / n);
        left %= n;

        for(int i = 0; left > 0; i++) {
            ans[i]++;
            left--;
        }

        return ans;
    }
};

File: 2155-find-missing-observations/2155-find-missing-observations.go
func missingRolls(rolls []int, mean int, n int) []int {
    m := len(rolls)
    goal := (m + n) * mean

    for _, v := range rolls {
        goal -= v
    }

    if goal < n || goal > 6 * n {
        return make([]int,0)
    }

    ans := make([]int, n, n)
    i := 0
    for i < goal % n {
        ans[i] = (goal / n) + 1
        i++
    }

    for i < n {
        ans[i] = (goal / n)
        i++
    }

    return ans
}